### PR TITLE
handle playback of background video when page loses focus (fixes issue #310)

### PR DESCRIPTION
**Pull request by cvan**

### CODE REVIEW

**Comment by gfodor** [comment]
LGTM! Lets sync on slack before merging so I can talk you through our deployment process. We perform continuous delivery of changes to master -- HEAD of master should always be what is live on the site.

**Comment by cvan** [comment]
@gfodor: sounds good! will sync with you on Monday on Slack. (did some play testing yesterday with a high-school student I mentor and @caseyyee, so I am filing issues as I remember them so I don’t forget. we were all really impressed by the stability and comms, and spent a good time in it - all three of us.)

File: src/react-components/home-root.js [hunk 1]
```diff
@@ -35,9 +35,39 @@ class HomeRoot extends Component {
...
+    this.loadHomeVideo();
   }
 
+  loadHomeVideo = () => {
+    const videoEl = document.querySelector("#background-video");
+    function initVideo() {
+      videoEl.playbackRate = 0.75;
+      videoEl.play();
+      let inIframe = false;
+      try {
```
Comment (netpro2k) [inline src/react-components/home-root.js]: Whats the deal with this?

File: src/react-components/home-root.js [hunk 2]
```diff
@@ -35,9 +35,39 @@ class HomeRoot extends Component {
...
+      }
+      function toggleVideo() {
+        // Play the video if the window/tab is visible (or is within an `<iframe>`).
+        if (document.hasFocus() || inIframe) {
+          videoEl.play();
+        } else {
+          videoEl.pause();
+        }
+      }
+      document.addEventListener("visibilitychange", toggleVideo);
```
Comment (netpro2k) [inline src/react-components/home-root.js]: Probably fine but just out of curiosity why do we need both of these? What cases do focus and blur catch that visiblitychange does not (and vis versa)?

File: src/react-components/home-root.js [hunk 3]
```diff
@@ -35,9 +35,39 @@ class HomeRoot extends Component {
...
+      window.addEventListener("focus", toggleVideo);
+      window.addEventListener("blur", toggleVideo);
+    }
+    if (videoEl.readyState >= videoEl.HAVE_FUTURE_DATA) {
```
Comment (cvan) [inline src/react-components/home-root.js]: I just have to take a second to lol at these state properties of [`HTMLMediaElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/readyState#Syntax)

File: src/react-components/home-root.js (hunk 2 above)
Comment (cvan) [reply src/react-components/home-root.js]: also, from the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API#Use_cases):

> Note: While `blur` and `focus` will tell you if the user switches windows, it doesn't necessarily mean it's hidden. Pages only become hidden when the user switches tabs or minimizes the browser window containing the tab.

**Pull request merged**